### PR TITLE
Fix bug when passing lambda to define-clojure-indent, closes #383

### LINKED ISSUE #383: Cannot pass lambdas to clojure-indent-function
## Expected behavior
The following should work in `define-clojure-indent`

`(trace-handlers (lambda (&rest _) 0))`

## Actual behavior
It fails to trigger the right case branch.

## Steps to reproduce the problem
```
(define-clojure-indent
  (trace-handlers (lambda (&rest _) 0)))
```

### clojure-mode version information
```
clojure-mode (version 5.3.0)
```

The problem seems [here](https://github.com/clojure-emacs/clojure-mode/blob/master/clojure-mode.el#L1238) where `functionp` returns `nil` as `lambda` is a special form.

### PR DESCRIPTION
**Pull request by arichiardi**
@Malabarba I added a test and it passes :+1: 
Also, I tried against my codebase with success!

Thanks a lot!

### CODE REVIEW

File: test/clojure-mode-indentation-test.el [hunk 1]
```diff
@@ -164,8 +164,19 @@ values of customisable variables."
...
+(check-indentation symbol-with-lambda
+  "
+(arsymbol
+ |one)"
   "
+(arsymbol
+|one)")
+
+(check-indentation form-with-metadata
+                   "
```
Comment (Malabarba) [inline test/clojure-mode-indentation-test.el]: Indent looks off here.

Comment (arichiardi) [inline test/clojure-mode-indentation-test.el]: Ok well, it seems aggressive mode was not disabled after all

**Comment by arichiardi** [comment]
It should be ok now, thanks for double checking :wink:

**Comment by Malabarba** [comment]
Great! Thanks for the tests.

**Pull request merged**